### ISSUE TITLE
Olivia Crashes on Fedora 31

### ISSUE THREAD

**Issue by alihoseiny** [keshavbhatt/olivia#88]
I installed Olivia today. Each time the program will close after several minutes. Here is the log has gotten from `dmesg` command:
```
olivia[39166]: segfault at 8 ip 00007fd7ec893ea2 sp 00007ffe49a054a0 error 4 in libQt5Widgets.so.5.9.5[7fd7ec706000+616000]
[24803.123489] Code: b4 af fb ff e9 53 ff ff ff e8 ea 85 fb ff 66 2e 0f 1f 84 00 00 00 00 00 55 53 89 f2 83 f2 01 89 f3 be 20 00 00 00 48 83 ec 08 <48> 8b 6f 08 0f b6 d2 e8 32 70 ff ff 48 83 c4 08 0f b6 f3 48 89 ef
[24979.181537] olivia[60375]: segfault at 8 ip 00007fadc836fea2 sp 00007ffd05e0a520 error 4 in libQt5Widgets.so.5.9.5[7fadc81e2000+616000]
[24979.181548] Code: b4 af fb ff e9 53 ff ff ff e8 ea 85 fb ff 66 2e 0f 1f 84 00 00 00 00 00 55 53 89 f2 83 f2 01 89 f3 be 20 00 00 00 48 83 ec 08 <48> 8b 6f 08 0f b6 d2 e8 32 70 ff ff 48 83 c4 08 0f b6 f3 48 89 ef
```

OS: Fedora 31
kernel info: 5.5.7-200.fc31.x86_64

**Comment by keshavbhatt** [comment]
Hi, thanks for reporting. 
Few things i would like to know :
How you installed Olivia ? 
Was the app running fine earlier ?
Thanks

**Comment by alihoseiny** [comment]
I've installed it using snapcraft. I installed it yesterday and it had the same behavior all the time.

**Comment by keshavbhatt** [comment]
when the crash happens ? at start or when you do something specific ?

**Comment by alihoseiny** [comment]
It happened while playing music or radio.

**Comment by keshavbhatt** [comment]
can you show me the output of 
```
snap connections olivia 
```

**Comment by alihoseiny** [comment]
Here is the output:
```
Interface              Plug                    Slot                            Notes
alsa                   olivia:alsa             -                               -
audio-playback         olivia:audio-playback   :audio-playback                 -
browser-support        olivia:browser-support  :browser-support                -
content[gtk-3-themes]  olivia:gtk-3-themes     gtk-common-themes:gtk-3-themes  -
content[icon-themes]   olivia:icon-themes      gtk-common-themes:icon-themes   -
content[qt551]         olivia:qt551            qt551:qt551                     -
content[sound-themes]  olivia:sound-themes     gtk-common-themes:sound-themes  -
desktop                olivia:desktop          :desktop                        -
desktop-legacy         olivia:desktop-legacy   :desktop-legacy                 -
gsettings              olivia:gsettings        :gsettings                      -
home                   olivia:home             :home                           -
mount-observe          olivia:mount-observe    -                               -
mpris                  -                       olivia:mpris                    -
network                olivia:network          :network                        -
network-bind           olivia:network-bind     :network-bind                   -
network-manager        olivia:network-manager  -                               -
opengl                 olivia:opengl           :opengl                         -
process-control        olivia:process-control  -                               -
unity7                 olivia:unity7           :unity7                         -
x11                    olivia:x11              :x11                            -
```

**Comment by keshavbhatt** [comment]
everything looks fine here, the issue seems to be with the code, somewhere with QWidget lib. i will try to find out the problem .
thanks for reporting.

**Comment by keshavbhatt** [comment]
Hi again, can you delete this folder and try running olivia again ?
`~/snap/olivia/current/.local/share/org.keshavnrj.ubuntu/Olivia/videoDownloadHistory`

**Comment by alihoseiny** [comment]
I've tried that and it did not crash for the last 3 hours.

**Comment by keshavbhatt** [comment]
cool, related fix is coming soon. 

thanks for updating :)

**Comment by keshavbhatt** [comment]
this is fixed in b39a2c4
